src/ln/channel.rs

```diff
@@ -2472,6 +2486,22 @@ impl Channel {
 			return Err(ChannelError::Close("Peer sent a garbage channel_reestablish"));
 		}
 
+		if msg.next_remote_commitment_number > 0 {
+			match msg.data_loss_protect {
+				OptionalField::Present(ref data_loss) => {
+					if chan_utils::build_commitment_secret(self.local_keys.commitment_seed, INITIAL_COMMITMENT_NUMBER - msg.next_remote_commitment_number + 1) != data_loss.your_last_per_commitment_secret {
+						return Err(ChannelError::Close("Peer sent a garbage channel_reestablish with secret key not matching the commitment height provided"));
+					}
+					if INITIAL_COMMITMENT_NUMBER - msg.next_remote_commitment_number < self.cur_local_commitment_transaction_number {
```

Comment:
I find this hard to read - can you use INITIAL_COMMITMENT_NUMBER - self.cur_... like the later if statements use? (to convert the values into incrementally-increasing values instead of decreasing values before the comparison).

Reply:
Done